build/patches/add-user-scripts.v5.patch

```diff
@@ -0,0 +1,9823 @@
++  // were not expecting.
++  void ScheduleAtDocumentStart(base::OnceClosure callback);
++
++  // Schedule a callback, to be run at the next RunScriptsAtDocumentEnd call.
++  void ScheduleAtDocumentEnd(base::OnceClosure callback);
++
++  // Schedule a callback, to be run at the next RunScriptsAtDocumentIdle call.
++  void ScheduleAtDocumentIdle(base::OnceClosure callback);
++
++ private:
++
++  void OnDestruct() override;
++
++  // The id of the tab the render frame is attached to.
++  int tab_id_;
++
++  // Callbacks to be run at the next RunScriptsAtDocumentStart notification.
++  std::vector<base::OnceClosure> document_element_created_callbacks_;
++
++  // Callbacks to be run at the next RunScriptsAtDocumentEnd notification.
++  std::vector<base::OnceClosure> document_load_finished_callbacks_;
++
++  // Callbacks to be run at the next RunScriptsAtDocumentIdle notification.
++  std::vector<base::OnceClosure> document_idle_callbacks_;
++
++  base::WeakPtrFactory<ExtensionFrameHelper> weak_ptr_factory_{this};
++
++  DISALLOW_COPY_AND_ASSIGN(ExtensionFrameHelper);
++};
++
++}  // namespace extensions
++
++#endif  // EXTENSIONS_RENDERER_EXTENSION_FRAME_HELPER_H_
+diff --git a/components/user_scripts/renderer/injection_host.cc b/components/user_scripts/renderer/injection_host.cc
+new file mode 100755
+--- /dev/null
++++ b/components/user_scripts/renderer/injection_host.cc
+@@ -0,0 +1,12 @@
++// Copyright 2015 The Chromium Authors. All rights reserved.
++// Use of this source code is governed by a BSD-style license that can be
++// found in the LICENSE file.
++
++#include "injection_host.h"
++
++InjectionHost::InjectionHost(const HostID& host_id) :
++    id_(host_id) {
++}
++
++InjectionHost::~InjectionHost() {
++}
+diff --git a/components/user_scripts/renderer/injection_host.h b/components/user_scripts/renderer/injection_host.h
+new file mode 100755
+--- /dev/null
++++ b/components/user_scripts/renderer/injection_host.h
+@@ -0,0 +1,42 @@
++// Copyright 2015 The Chromium Authors. All rights reserved.
++// Use of this source code is governed by a BSD-style license that can be
++// found in the LICENSE file.
++
++#ifndef EXTENSIONS_RENDERER_INJECTION_HOST_H_
++#define EXTENSIONS_RENDERER_INJECTION_HOST_H_
++
++#include "base/macros.h"
++#include "../common/host_id.h"
++#include "url/gurl.h"
++
++namespace content {
++class RenderFrame;
++}
++
++// An interface for all kinds of hosts who own user scripts.
++class InjectionHost {
++ public:
++  InjectionHost(const HostID& host_id);
++  virtual ~InjectionHost();
++
++  // Returns the CSP to be used for the isolated world. Currently this only
++  // bypasses the main world CSP. If null is returned, the main world CSP is not
++  // bypassed.
++  virtual const std::string* GetContentSecurityPolicy() const = 0;
++
++  // The base url for the host.
++  virtual const GURL& url() const = 0;
++
++  // The human-readable name of the host.
++  virtual const std::string&
```

Comment:
How come this is removed?

Reply:
I do not understand. Do you mean `CanExecuteOnFrame`?
That call only makes sense for extensions, because it requires authorization with another mojo call to the `HostID`.
However I check and give you a more certain answer

Reply:
No, I mean `assert superType.indexOf('/') == -1;` (which is commented); perhaps you copied it from an older Bromite patch?

Reply:
you missed my comment the line above

```
//assert removed from upstream <-- my comment
//assert superType.indexOf('/') == -1;
```

sorry i forgot to take it off